### PR TITLE
Update webgpu.h with dawn webgpu_headers_gen

### PR DESCRIPTION
**Pull request by shrekshao**
Update with dawn `webgpu_headers_gen` [@b0987d](https://dawn.googlesource.com/dawn/+/b0987d6e46ae4be58b009af9c242d4c841519f05)

### CODE REVIEW

File: webgpu.h [hunk 1]
```diff
@@ -239,6 +239,7 @@ typedef enum WGPUFeatureName {
     WGPUFeatureName_TextureCompressionETC2 = 0x00000007,
     WGPUFeatureName_TextureCompressionASTC = 0x00000008,
     WGPUFeatureName_IndirectFirstInstance = 0x00000009,
+    WGPUFeatureName_DawnNative = 0x000003EC,
```
Comment (austinEng) [inline webgpu.h:242]: this looks like it's Dawn internal and shouldn't be here

File: webgpu.h [hunk 2]
```diff
@@ -903,22 +932,11 @@ typedef struct WGPUSurfaceDescriptorFromWindowsHWND {
...
-typedef struct WGPUSurfaceDescriptorFromWaylandSurface {
-    WGPUChainedStruct chain;
-    void * display;
-    void * surface;
-} WGPUSurfaceDescriptorFromWaylandSurface;
-
-typedef struct WGPUSurfaceDescriptorFromAndroidNativeWindow {
-    WGPUChainedStruct chain;
-    void * window;
-} WGPUSurfaceDescriptorFromAndroidNativeWindow;
```
Comment (austinEng) [inline webgpu.h:921]: shouldn't drop these. Should add them to Dawn instead

File: webgpu.h (hunk 1 above)
Comment (shrekshao) [reply webgpu.h:242]: Shall we remove "native" tag for this feature name?
https://source.chromium.org/chromium/chromium/src/+/main:third_party/dawn/dawn.json;l=1348?q=%22dawn%22,%20%22native%22%20f:&ss=chromium

File: webgpu.h [hunk 3]
```diff
@@ -710,6 +718,25 @@ typedef struct WGPUConstantEntry {
     double value;
 } WGPUConstantEntry;
 
+typedef struct WGPUDawnCacheDeviceDescriptor {
```
Comment (austinEng) [inline webgpu.h:721]: these too are Dawn internal

**Review by austinEng — APPROVED** [review]

**Review by kvark — APPROVED** [review]

File: webgpu.h [hunk 4]
```diff
@@ -346,7 +346,7 @@ typedef enum WGPUSType {
     WGPUSType_Invalid = 0x00000000,
     WGPUSType_SurfaceDescriptorFromMetalLayer = 0x00000001,
     WGPUSType_SurfaceDescriptorFromWindowsHWND = 0x00000002,
-    WGPUSType_SurfaceDescriptorFromXlibWindow = 0x00000003,
+    WGPUSType_SurfaceDescriptorFromXlib = 0x00000003,
```
Comment (rajveermalviya) [inline webgpu.h:349]: I think in [#140](https://github.com/webgpu-native/webgpu-headers/pull/140#discussion_r778264107) it was discussed to rename `Xlib`. So I think it's okay to keep this `XlibWindow` ?

Comment (shrekshao) [reply webgpu.h:349]: Oops I actually changed dawn to fit naming in webgpu-headers just for this. Let me change things back

**Pull request merged**